src/main/java/org/checkerframework/specimin/MustImplementMethodsVisitor.java

```diff
@@ -63,15 +64,16 @@ public Visitable visit(MethodDeclaration method, Void p) {
     // implementing required methods from interfaces in the target code,
     // but unfortunately I think it's the best that we can do here. (@Override
     // is technically optional, but it is widely used.)
-    if (isPreservedAndAbstract(overridden)
+    if (ancestorMethodPreservedAndAbstract(method)
         || (overridden == null && overridesAnInterfaceMethod(method))) {
       ResolvedMethodDeclaration resolvedMethod = method.resolve();
-      Set<String> returnAndParamTypes = new HashSet<>();
+      Map<String, ResolvedType> returnAndParamTypes = new HashMap<>();
```

Comment:
I think the values in this map also need to include thrown exceptional types. The new code exposes a problem caused by not doing that in CF-6060, which it looks like is why that integration test fails.